public/icvar.h

```diff
@@ -11,21 +11,456 @@
+
+class ConCommandHandle
+{
+public:
+	ConCommandHandle( uint16_t index = -1, uint32_t handle = -1) :
+	m_concommandIndex(index),
+	m_handleIndex(handle)
+	{}
+
+	bool IsValid( ) const { return m_concommandIndex != 0xFFFF; }
+	void Invalidate( )
+	{
+		m_concommandIndex = 0xFFFF;
+		m_handleIndex = 0x0;
+	}
+
+	uint32_t GetConCommandIndex() const { return m_concommandIndex; }
+	uint32_t GetIndex() const { return m_handleIndex; }
+
+private:
+	uint32_t m_concommandIndex;
+	uint32_t m_handleIndex;
+};
+
+static const ConCommandHandle INVALID_CONCOMMAND_HANDLE = ConCommandHandle();
+
+//-----------------------------------------------------------------------------
+// Purpose: Internal structure of ConVar objects
+//-----------------------------------------------------------------------------
+enum EConVarType : int16_t
+{
+	EConVarType_Invalid = -1,
+	EConVarType_Bool,
+	EConVarType_Int16,
+	EConVarType_UInt16,
+	EConVarType_Int32,
+	EConVarType_UInt32,
+	EConVarType_Int64,
+	EConVarType_UInt64,
+	EConVarType_Float32,
+	EConVarType_Float64,
+	EConVarType_String,
+	EConVarType_Color,
+	EConVarType_Vector2,
+	EConVarType_Vector3,
+	EConVarType_Vector4,
+	EConVarType_Qangle,
+	EConVarType_MAX
+};
+
+template<typename T>
+constexpr EConVarType TranslateConVarType();
+
+template<> constexpr EConVarType TranslateConVarType<bool>( void )		{ return EConVarType_Bool; }
+template<> constexpr EConVarType TranslateConVarType<int16_t>( void )	{ return EConVarType_Int16; }
+template<> constexpr EConVarType TranslateConVarType<uint16_t>( void )	{ return EConVarType_UInt16; }
+template<> constexpr EConVarType TranslateConVarType<int32_t>( void )	{ return EConVarType_Int32; }
+template<> constexpr EConVarType TranslateConVarType<uint32_t>( void )	{ return EConVarType_UInt32; }
+template<> constexpr EConVarType TranslateConVarType<int64_t>( void )	{ return EConVarType_Int64; }
+template<> constexpr EConVarType TranslateConVarType<uint64_t>( void )	{ return EConVarType_UInt64; }
+template<> constexpr EConVarType TranslateConVarType<float>( void )		{ return EConVarType_Float32; }
+template<> constexpr EConVarType TranslateConVarType<double>( void )	{ return EConVarType_Float64; }
+template<> constexpr EConVarType TranslateConVarType<const char*>( void ){ return EConVarType_String; }
+template<> constexpr EConVarType TranslateConVarType<Color>( void )		{ return EConVarType_Color; }
+template<> constexpr EConVarType TranslateConVarType<Vector2D>( void )	{ return EConVarType_Vector2; }
+template<> constexpr EConVarType TranslateConVarType<Vector>( void )	{ return EConVarType_Vector3; }
+template<> constexpr EConVarType TranslateConVarType<Vector4D>( void )	{ return EConVarType_Vector4; }
+template<> constexpr EConVarType TranslateConVarType<QAngle>( void )	{ return EConVarType_Qangle; }
+template<> constexpr EConVarType TranslateConVarType<void*>( void )		{ return EConVarType_Invalid; }
+
+template<typename T>
+struct ConVarData_t;
+
+struct ConVarBaseData_t

```

Comment:
While it's not too late to change the names, how about change this to a class and remove the ``_t``?